### ISSUE TITLE
Bleach The Heat Soul 7 become slower after removing the software skinning

### ISSUE THREAD

**Issue by Jinzombie** [hrydgard/ppsspp#10775]
What happens? Bleach The Heat Soul 7 become slower after removing software skinning in graphics performance

What *should* happen?

What hardware and operating system are you running PPSSPP on? GPU might matter if it's a graphical issue.

Phone Specs:
Mali-450
1.4Ghz
Octacore
2GB RAM

**Comment by hrydgard** [comment]
Software skinning is still there, it's what's always used now. But maybe you had it unchecked before.

How big is the performance difference?

**Comment by Jinzombie** [comment]
Yes I unchecked it before because it cause lag 
From 30 FPS now I only Got 20 FPS with the same settings

Sorry for my english ✌

**Comment by Jinzombie** [comment]
Im using the latest ppsspp build 1.5.4-729

**Comment by unknownbrackets** [comment]
Adding milestone because this is a regression.  I wonder how it impacted performance here so much, though...

-[Unknown]

**Comment by hrydgard** [comment]
Darn. In my testing I didn't find much that was impacted at all...  Really quite surprising.

Just to check, @Jinzombie , you compared versions just before and after the removal of the skinning option?

**Comment by SamueldeSena** [comment]
Also I have this problem in some games and I also want to emphasize that in the builds most resent with api vulkan games that rada and 60FPS with for example god of war this closing the emulator I have a question has possibility of opicao software to return in the emulator and ppsspp settings could have an option that improve the latency of the audio better than already has an option that diminishes the hiss or an option to have a fixed amount of FPS of this already I thank to

**Comment by hrydgard** [comment]
@SamueldeSena  Sorry it's really hard to figure out what you are saying, please write more clearly. Also that sounds like several separate issues, and don't belong here in this Bleach thread.

**Comment by Jinzombie** [comment]
@hrydgard works fine in version 1.5.4-630 since I can turn off the software skinning in the option but in version 1.5.4-643 the problem starts no option to turn it off also I only get low FPS during battle.

**Comment by hrydgard** [comment]
@Jinzombie  ok, thanks for confirming.

**Comment by Jinzombie** [comment]
@hrydgard  your welcome :)

**Comment by unknownbrackets** [comment]
Could you try exporting a GE debugger dump on PC?  I know you're experiencing the issue on Android, but we only support creating a GE debugger dump on Windows right now.

To do this, open the game and select Debug -> GE debugger..., then when it's displaying the scene, press Record in the top right.  After a second or so, it'll finish and save a trace of the drawing activity.

After that, check the memstick/PSP/SYSTEM/DUMP folder and it'll have created a file named something like "ULES12345_0000.ppdmp".  You can zip that and then drag and drop it into a reply here.

I'm hoping that'll help us reproduce the issue.

-[Unknown]

**Comment by unknownbrackets** [comment]
If the GE dump is hard to get - do you know if there's any demo version of this game that has the same behavior?

-[Unknown]

**Comment by Jinzombie** [comment]
@Unknownbrackets I couldn't find any demo version of Bleach The Heat Soul 7 maybe I should try the GE debugger option

**Comment by Jinzombie** [comment]
I try other games that don't use software skinning and I finally found it in Monster Hunter 3rd same as Bleach The Heat Soul 7 they become slower after removing the option of software skinning in the graphics performance.
here's the link of the Demo Version http://download.digiex.net/Consoles/PSP/Demos/MonsterHunterPortable3rdDemoJPN.rar
Hoping this can help 😇

**Comment by Jinzombie** [comment]
@unknownbrackets I tested Monster Hunter 3RD that has similar settings in BTH7 and I find out that is also become slow here's the Demo Version Link of this game hope this can help
http://download.digiex.net/Consoles/PSP/Demos/MonsterHunterPortable3rdDemoJPN.rar

*Sorry for my bad english

**Comment by Jinzombie** [comment]
@unknownbrackets Really hard to get the GE DEBUGGER always crash the ppsspp emulator in my pc 32bit :(

**Comment by Jinzombie** [comment]
Maybe this could help
Problem signature:
  Problem Event Name:	BEX
  Application Name:	PPSSPPWindows.exe
  Application Version:	1.5.4.837
  Application Timestamp:	5ac7ec42
  Fault Module Name:	StackHash_0a9e
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	00000000
  Exception Offset:	00000000
  Exception Code:	c0000005
  Exception Data:	00000008
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by unknownbrackets** [comment]
That's strange, it doesn't crash for me even on 32-bit... maybe something doesn't support Windows XP.

I'll see if I can look at the Monster Hunter 3rd Demo soon.  Is there any particular area that's a good example?

-[Unknown]

**Comment by Jinzombie** [comment]
@unknownbrackets In the village area its laggy because Software Skinning is always on but only in the village.Unlike Bleach The Heat Soul 7 during fighting scence in all area/stage its very lag on my android phone :(

**Comment by hrydgard** [comment]
OK, guess I underestimated the boost a few games got from it, considering adding back support for hardware skinning. Will be a nasty merge, heh.

**Comment by zminhquanz** [comment]
I think you can keep and add back Software Skinning checkbox , because some game hate this

**Comment by hrydgard** [comment]
@zminhquanz @Jinzombie You can now disable software skinning again, resulting in hardware (shader) skinning. Please confirm that performance is back to the expected level :)

**Comment by Jinzombie** [comment]
@hrydgard YES! 😀 the performance is back of BTHS7 and other game that don't need SS.
Thank You Very Much For Your Hardwork May God Bless You 😇

**Comment by unknownbrackets** [comment]
Hopefully we can figure out what's causing the performance problem and get the best of both worlds at some point.

-[Unknown]

**Comment by Gamemulatorer** [comment]
It's been a long time since this issue I posted 🤣
But on my new phone Redmi Note 9 enabling or disabling the software skinning does not affect the performance on Bleach The Heat Soul 7 and MH 3rd Portable HD.

I think GLES2 devices only suffer this issue and not OGL3+ on android.

**Comment by hrydgard** [comment]
Yeah on modern devices, either way is fast enough that it doesn't matter.

**Comment by Gamemulatorer** [comment]
But on my ancient phone (Mali-450) this option hurt the performance on BTHS7 and both MH games.
Maybe some useful debug?
'''
[09-09 00:42:04.242 914:969 D/ViewRootImpl]
hardware acceleration is disabled, fakeHwAccelerated = true, HardwareRenderer.sRendererDisabled = false, forceHwAccelerated = false, HardwareRenderer.sSystemRendererDisabled = false, this = ViewRoot{3a84ae2c Starting org.ppsspp.ppsspp,ident = 84}
[09-09 00:42:04.738 6116:6116 I/PPSSPP]
EARLY: ThreadManager::Init(compute threads: 8, all: 12)
[09-09 00:42:05.021 6116:6116 I/PPSSPPNativeActivity]
OpenGL ES 2.0 detected
'''
![Screenshot_2021-09-09-00-43-20](https://user-images.githubusercontent.com/37603562/132555207-dce63aea-c462-41ee-b2d4-df3e0da411a1.png)

**Comment by hrydgard** [comment]
Nah, we kept both options just for the case of old devices that prefer one or the other. Not much to do.
